GiniVision/Classes/Networking/APIService.swift

```diff
@@ -0,0 +1,167 @@
+//
+//  DocumentService.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 2/14/18.
+//
+
+import UIKit
+import Gini_iOS_SDK
+
+let GINIAnalysisManagerDidReceiveResultNotification = "GINIAnalysisManagerDidReceiveResultNotification"
+let GINIAnalysisManagerDidReceiveErrorNotification  = "GINIAnalysisManagerDidReceiveErrorNotification"
+let GINIAnalysisManagerResultDictionaryUserInfoKey  = "GINIAnalysisManagerResultDictionaryUserInfoKey"
+let GINIAnalysisManagerErrorUserInfoKey             = "GINIAnalysisManagerErrorUserInfoKey"
+let GINIAnalysisManagerDocumentUserInfoKey          = "GINIAnalysisManagerDocumentUserInfoKey"
+
+public typealias Extraction = GINIExtraction
+typealias DocumentAnalysisCompletion = (([String: Extraction]?, GINIDocument?, Error?) -> Void)
+
+final class APIService {
+    
+    fileprivate var cancelationToken: CancelationToken?
+    var document: GINIDocument?
+    var error: Error?
+    var giniSDK: GiniSDK?
+    var isAnalyzing = false
+    var result: [String: Extraction]?
+
+    enum AnalysisError: Error {
+        case documentCreation
+        case unknown
+    }
+    
+    func cancelAnalysis() {
+        cancelationToken?.cancel()
+        result = nil
+        document = nil
+        error = nil
+        isAnalyzing = false
+    }
+    
+    init(sdk: GiniSDK?) {
+        self.giniSDK = sdk
+    }
+    
+    func analyzeDocument(withData data: Data,
+                         cancelationToken token: CancelationToken,
+                         completion: DocumentAnalysisCompletion?) {
+        print("🔎 Started document analysis with size \(Double(data.count) / 1024.0)")
+        
+        cancelAnalysis()
+        cancelationToken = token
+        isAnalyzing = true
+        
+        let manager = giniSDK?.documentTaskManager
+        let fileName = "fileName"
```

Comment:
Why do you need a file name? And why is it always the same?

Reply:
`createDocumentWithFilename` methods require a fileName. The `fileName` values it was just for testing purposes, It could make more sense to set it with the timestamp or uuid.

Reply:
Yeah, I see that. It might make sense to se if `createDocumentWithFilename ` really uses the file name for something meaningful and if not, to remove it. I assume that we currently don't need more than one file. But that might change with multipage.

Reply:
It is added to the query, but it will be change when we implement the multipage.